services/mbot_oled_display.py

```diff
@@ -30,20 +48,21 @@ def get_wlan0_ip():
         # Use regular expressions to find the IP address in the output
         ip_match = re.search(r'inet ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)', command_output)
         if ip_match:
-            ip_address = ip_match.group(1)
-            return ip_address
+            return ip_match.group(1)
         else:
-            return None
+            return "IP Not Found"
```

Comment:
I think returning `None` is fine here, because it matches the description of the function. It's called `get_wlan0_ip()`, indicating that it gets the IP, so the fact that it returns either the IP as a string or a string error message is unexpected from a development point of view. Whoever uses this function should then be responsible for checking whether the function returns `None` and handling it in a way that makes sense for the application (e.g. `if ip is None: msg = "No IP"`)